src/Microsoft.Tye.Core/ApplicationFactory.cs

```diff
@@ -21,227 +21,257 @@ public static async Task<ApplicationBuilder> CreateAsync(OutputContext output, F
                 throw new ArgumentNullException(nameof(source));
             }
 
-            var config = ConfigFactory.FromFile(source);
-            config.Validate();
-            var builder = new ApplicationBuilder(source, config.Name ?? source.Directory.Name.ToLowerInvariant());
-            if (!string.IsNullOrEmpty(config.Registry))
-            {
-                builder.Registry = new ContainerRegistry(config.Registry);
-            }
+            var queue = new Queue<ConfigApplication>();
+            var visited = new HashSet<string>();
```

Comment:
This needs a comparer - on windows at least some APIs normalize case and some preserve it. Depending on the API used you might see the same file twice with two casings.